src/protobuf/bmv2/table.proto

```diff
@@ -0,0 +1,43 @@
+syntax = "proto3";
+
+import "google/protobuf/struct.proto";
+
+import "src/protobuf/bmv2/primitives.proto";
+
+message Action {
+  string name = 1;
+  int32 id = 2;
+  repeated RuntimeData runtime_data = 3;
+  repeated google.protobuf.Struct primitives = 4;  
+}
+
+message Pipeline {
+  string name = 1;
+  int32 id = 2;
+  SourceInfo source_info = 3;
+  string init_table = 4;
+  repeated Table tables = 5;
+  // TODO action_profiles
+  // TODO conditionals
+}
+
+message Table {
+  string name = 1;
+  int32 id = 2;
+  SourceInfo source_info = 3;
+  repeated TableKey key = 4;
+  string match_type = 5; // make into an enum
```

Comment:
Should this be a TODO? At Google we generally start todo's with "TODO(username): ...".